network/network.go

```diff
@@ -115,8 +117,21 @@ func (n *Network) Configure(config core.ServerConfig) error {
 		return fmt.Errorf("unable to migrate DAG: %w", err)
 	}
 
+	// NATS
+	natsAddr := fmt.Sprintf("%s:%d", n.config.NatsHostname, n.config.NatsPort)
+
+	conn, err := nats.Connect(natsAddr, nats.RetryOnFailedConnect(true), nats.Timeout(time.Duration(n.config.NatsTimeout)*time.Second))
```

Comment:
should this be done here, or in some other engine (I believe we have the 'event bus' engine?)

Reply:
You can actually use an external NATS instance if you want. This is only the client NATS connection so in my opinion it should definitely be here

Reply:
there is a `events.Connect` convenience method (with different args though) that could be put back in action. Could also be a nice coupling the the NatsConfig struct mentioned earlier?

Reply:
That's true, will do 👍